src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -32,35 +32,43 @@ def fetch_alert(self, **notification):
 class PrometheusRESTController(RESTController):
     def prometheus_proxy(self, method, path, params=None, payload=None):
         # type (str, str, dict, dict)
-        user, password, cert_file = self.get_access_info('prometheus')
-        verify = cert_file.name if cert_file else Settings.PROMETHEUS_API_SSL_VERIFY
+        user, password, ca_cert_file, cert_file, key_file = self.get_access_info('prometheus')
+        verify = ca_cert_file.name if ca_cert_file else Settings.PROMETHEUS_API_SSL_VERIFY
+        cert = (cert_file.name, key_file.name) if cert_file and key_file else None
         response = self._proxy(self._get_api_url(Settings.PROMETHEUS_API_HOST),
                                method, path, 'Prometheus', params, payload,
-                               user=user, password=password, verify=verify)
-        if cert_file:
-            cert_file.close()
-            os.unlink(cert_file.name)
+                               user=user, password=password, verify=verify,
+                               cert=cert)
+        for f in [ca_cert_file, cert_file, key_file]:
+            if f:
+                f.close()
+                os.unlink(f.name)
```

Comment:
this is duplicated below as well so shall we have a separate function to do this?

Reply:
fixed